### ISSUE TITLE
Sometimes, a single, simple test will fail under multi-threaded test running

### ISSUE THREAD

**Issue by msooseth** [ethereum/hevm#556]
Sometimes, a single, simple test will fail under multi-threaded test running. Specifically:

```
cabal run -f devel test -- -p '/access-out-of-bounds-array/'
```

if ran 10x or so on an x64, it will fail. Both `solc` and the SMT solver (`z3`?) are being communicated via channels, so it makes no sense to me, it shouldn't be a file write race/hazard, as the case was with #555 . Something is odd.

**Comment by blishko** [comment]
Hmm, on my arm Mac, there was no failure (ran it 50 times).

**Comment by msooseth** [comment]
OK, so this has been nailed down to 

```
-        (a, b) <- liftIO $ concurrently evalLeft evalRight
+        a <- liftIO $ evalLeft
+        b <- liftIO $ evalRight
```

Will be fixed once #555 is merged. Sad day.

**Comment by blishko** [comment]
This was fixed by #555 and #560.
